### ISSUE TITLE
Revert to `bundle` syntax in CLI

### ISSUE THREAD

**Issue by corang** [defenseunicorns/uds-cli#73]
This is something I feel rather strongly about and not a case of "less is more". While currently the only functionality of uds-cli is for managing bundles, I think it's rather agreed upon that there will be other functionality integrated into uds-cli as time progresses. This is implied [here](https://defense-unicorns.slack.com/archives/C04T24A54LU/p1695698269211039?thread_ts=1695678398.886819&cid=C04T24A54LU) by @jeff-mccoy.

I am of the opinion that not enough time/input was allowed for the decision of deprecating the `bundle` syntax given that it seems that most of the feedback from the people using uds-cli was, at least initially, to not do this.

The arguments that I've seen for this seem to just boil down to "less typing is always better" and "less words is better ux". While there are cases where this is good I don't think the difference between `uds b c` and `uds c` (you'd be using the shorthand if character count matters to you) is worth it. I also don't think removing the `bundle` word is better ux. There have been a few examples of other applications given as basis for why uds-cli should remove the bundle noun (`helm install` , `tf apply`) but I don't think those examples have the same use case as uds-cli.

This part is hard to articulate but those applications work like that because they are single purpose, helm installs helm charts, terraform applies terraform. While uds-cli is single purpose today, as I referenced above, I'm almost positive it will not be, and even from a naming perspective it doesn't make sense. `helm install` works because you're installing a "helm", `tf apply` works because you're applying terraform, `uds create` is not intuitive and doesn't make sense because you're not "creating a uds" you're creating a bundle. If the sole purpose of this application was for bundles and was named appropriately I wouldn't have any issues with it, but as it stands I think at best this change will be reverted relatively soon once any other functionality is added. And at worst if more functionality is added and this syntax stays it will only serve to confuse end users and make the application less appealing.

**Comment by mjnagel** [comment]
> `helm install` works because you're installing a "helm"

I'd push back on this - `helm install` installs a chart, so its almost 1:1 the same scenario as `uds create` creating a bundle. I don't know that I could make the same case with terraform - so that's a fair example as best I can tell.

I don't have a strong opinion on this, but I am curious what this "other functionality" is that would necessitate clarifying its a bundle action. I guess I was expecting this would always just work on bundles, but there may be additional things that could be pulled into bundles (i.e. non-zarf things like IaC)? Maybe the point you're making is that we don't really know though so why risk it...

**Comment by Racer159** [comment]
My 2 :coin:s would be that this could / should be driven by an ADR (probably not on this particular issue since it is small by itself) but having a base uds-cli / bundle ADR on what they are and what they are planned to contain would allow people to see more decision criteria / more of the vision / drive behind uds-cli and bundles.

Fun fact `zarf` actually started with a command hierarchy but also had shortcut commands that were removed early on (before Zarf started doing ADRs).  Zarf still has deprecation warnings in it's CLI almost 2 years on:

![image](https://github.com/defenseunicorns/uds-cli/assets/3977569/80603092-7e7b-49b7-8c0e-7c4a5ae65702)

https://github.com/defenseunicorns/zarf/blob/11f6173b69678399abce87c0a43abaacb22f516a/cli/cmd/root.go#L26

**Comment by Noxsios** [comment]
@mjnagel I agree, I think it depends upon what will necessitate a `bundle`, as there is not a clear definition / ADR on that as of right now. Is `bundle` the _only_ thing that UDS CLI will orchestrate? In order to update a single application in a bundle, are users going to re-build the bundle and re-deploy, or should UDS CLI support deploying Zarf packages and merging them into an existing bundle?

Even in Zarf there is a little bit of disrepency, `zarf connect` is _technically_ `zarf package connect`.

**Comment by anthonywendt** [comment]
Just my 2 cents. 

I feel that it is premature to remove clarity from such a naval tool. For example something like `helm install`, the action it performs on what is widely known as a helm chart has history. It's intuitive and it feels natural for chart to be implied when using the helm tool.

What is UDS-CLI? What is UDS to the world? At first glance with a little digging I find UDS is an acronym for Unicorn Delivery Service. Okay cool, that to me sounds like maybe UDS-CLI is a multi-purpose command line interface tool that makes software delivery much easier for me.  What are all the cool things it can help me do? Well, it has one purpose and that is to create and deploy these things called Bundles which are well orchestrated collections of zarf packages. It also inherits a lot of zarf functionality. Well what is zarf? Why isn't this just a feature of zarf if thats all it does, someone may think to themselves? Its not the multi purpose Unicorn Delivery Service tool I initially assumed.

That is just an example of the thoughts that would go around in my brain when coming across this tool. Implying bundle just feels premature and possibly handcuffs the potential of the tool. That is all.

**Comment by jeff-mccoy** [comment]
Hello folks, as product architect, I approved this change when it happened, I think @Racer159 is right an ADR would be needed to change this going forward. I'll give you my take on why I am fine with it:

- UDS CLI will have more behavior than just bundles, but we dont know what or when 
- But like Helm, the primary thing will be the bundle...the only reason anyone feels this normal is because helm made it so for that tool, and that's where we are with UDS-CLI, it's normal if we make it normal
- For additional things there may be a tree of subcommands pushed lower, similar to the `zarf tools <command>` pattern, but we don't know what those will be yet
- In the big picture, UDS CLI will be firstly a bundle handler and secondly, an view into the UDS Runtime Engine (name tbd), for which there will also be a UDS UI, both the CLI and UI will call effectively the same REST (tbd) endpoints. The only real in-CLI business logic will be that which must run locally, i.e. a `zarf package <COMMAND>` equivalent for bundles. 
- No final decision on miscellaneous tools, but I suspect it might end up being something like kubectl plugins for those things.

FWIW, I originally had both `zarf package create` + `zarf zarf.yaml` and `zarf package deploy` + `zarf package.tar.gz` so you could shorthand--but lost that argument (can't seem to find it now)

**Comment by UncleGedd** [comment]
Hey folks really appreciate all the discussion. We're making the call to keep the `uds <action>` syntax, with no `bundle` keyword. As for additional ADRs on the definition/purpose of a bundle, I'm all for sharing more words/context, but am admittedly unsure about what the content of that ADR should be (but that's for a different PR!)
